server/src/main/java/org/opensearch/index/mapper/NestedPathFieldMapper.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.mapper;
+
+import org.apache.lucene.document.Field;
+import org.apache.lucene.document.FieldType;
+import org.apache.lucene.index.IndexOptions;
+import org.apache.lucene.index.Term;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.TermQuery;
+import org.apache.lucene.util.BytesRef;
+import org.opensearch.Version;
+import org.opensearch.index.query.QueryShardContext;
+import org.opensearch.search.lookup.SearchLookup;
+
+import java.util.Collections;
+
+public class NestedPathFieldMapper extends MetadataFieldMapper {
+    // OpenSearch version 2.0 removed types; this name is used for bwc
+    public static final String LEGACY_NAME = "_type";
+    public static final String NAME = "_nested_path";
```

Comment:
Just an opinion, `_nested_path` looks too verbose, may be just `_path`?

Reply:
I ended up sticking with `_nested_path` because it's more descriptive when analyzing lucene segments (which has no object->field relationship). The fieldType is easily identifiable as the path for a nested field when just reading the field name.

Reply:
:+1: